sipyco/test/test_pyon_plugin.py

```diff
@@ -0,0 +1,58 @@
+import dataclasses
+
+import pluggy
+import pytest
+
+import sipyco
+import sipyco.hookspecs as hookspecs
+import sipyco.plugins as plugin
+import sipyco.pyon as pyon
+
+
+@dataclasses.dataclass
+class Point:
+    x: float
+    y: float
+
+
+class TestPyonPlugin:
+    @sipyco.hookimpl
+    def sipyco_pyon_encode(value, pretty=False):
+        if isinstance(value, Point):
+            return repr(value)
+
+    @sipyco.hookimpl
+    def sipyco_pyon_decoders():
+        return [("Point", Point)]
+
+
+def test_pyon_plugin_fail_without_plugin():
+    with pytest.raises(TypeError):
+        pyon.encode(Point(3, 4))
+
+
+def pyon_extra_plugin():
+    pm = pluggy.PluginManager("sipyco")
+    pm.add_hookspecs(sipyco.hookspecs)
+    pm.register(pyon)
+    pm.load_setuptools_entrypoints("sipyco")
```

Comment:
Should probably change this to ``pm = plugin.get_plugin_manager()``, then just add the register() at the end?